tests/test_network.ts

```diff
@@ -0,0 +1,75 @@
+
+
+import type { Test } from "../src/actions";
+
+export const NetworkDisconnectTest: Test = {
+  client: {
+    actions: [
+      { type: "invoke", id: "1", proc: "kv.set", payload: { k: "foo", v: 42 } },
+      { type: "invoke", id: "2", proc: "kv.set", payload: { k: "foo", v: 43 } },
+      { type: "disconnect_network" },
+      { type: "connect_network" },
+      { type: "invoke", id: "3", proc: "kv.set", payload: { k: "abc", v: 1 } },
+    ],
+    expectedOutput: [
+      { id: "1", status: "ok", payload: 42 },
+      { id: "2", status: "ok", payload: 43 },
+      { id: "3", status: "ok", payload: 1 },
+    ],
+  }
+}
+
+
+export const BufferRequestTest: Test = {
+  client: {
+    actions: [
+      { type: "invoke", id: "1", proc: "kv.set", payload: { k: "foo", v: 42 } },
+      { type: "disconnect_network" },
+      { type: "invoke", id: "2", proc: "kv.set", payload: { k: "foo", v: 43 } },
+      { type: "invoke", id: "3", proc: "kv.set", payload: { k: "abc", v: 1 } },
+      { type: "connect_network" },
+    ],
+    expectedOutput: [
+      { id: "1", status: "ok", payload: 42 },
+      { id: "2", status: "ok", payload: 43 },
+      { id: "3", status: "ok", payload: 1 },
+    ],
+  }
+}
+
+export const SubscriptionDisconnectTest: Test = {
+  client: {
+    actions: [
+      { type: "invoke", id: "1", proc: "kv.set", payload: { k: "foo", v: 42 } },
+      { type: "invoke", id: "2", proc: "kv.watch", payload: { k: "foo" } },
+      { type: "wait", ms: 800 },
+      { type: "disconnect_network" },
+      { type: "invoke", id: "3", proc: "kv.set", payload: { k: "abc", v: 1 } },
+      { type: "invoke", id: "4", proc: "kv.set", payload: { k: "foo", v: 43 } },
+    ],
+    expectedOutput: [
+      { id: "1", status: "ok", payload: 42 },
+      { id: "2", status: "ok", payload: 42 }, // watch should return the current value
+    ],
+  },
+};
+
+export const SubscriptionReconnectTest: Test = {
+  client: {
+    actions: [
+      { type: "invoke", id: "1", proc: "kv.set", payload: { k: "foo", v: 42 } },
+      { type: "invoke", id: "2", proc: "kv.watch", payload: { k: "foo" } },
+      { type: "disconnect_network" },
+      { type: "invoke", id: "3", proc: "kv.set", payload: { k: "abc", v: 1 } },
+      { type: "invoke", id: "4", proc: "kv.set", payload: { k: "foo", v: 43 } },
+      { type: "connect_network" },
+    ],
+    expectedOutput: [
+      { id: "1", status: "ok", payload: 42 },
+      { id: "2", status: "ok", payload: 42 }, // watch should return the current value
+      { id: "3", status: "ok", payload: 1 },
+      { id: "2", status: "ok", payload: 43 },
+      { id: "4", status: "ok", payload: 43 },
+    ],
+  },
+};
```

Comment:
let's do a test with 2 clients where one is connected and the other is disconnected

Reply:
Done